### PR TITLE
53: Create account page

### LINKED ISSUE #53: Implement create account page
**Description:**

- Create a new component that contains text input fields that the user can enter all of the required information for an account, which should be navigated from the login page link
- Validate that valid data is present when the user tries to submit 
- Make calls to the user service to create an account using the valid data
- If the account creation was successful, log the user into the app
- Write tests for the new components and the new flow

**Acceptance Criteria:**

- [x] Able to navigate to the new component from the login page
- [x] Input is validated (All fields filled out and UMNetId hasn't previously been used to register an account)
- [x] When a user enters valid data and submits, the account is created successfully and the user is logged into the app 

Blocked by #43  #44

### PR DESCRIPTION
**Pull request by ZackHolmberg**
closes #53

### CODE REVIEW

File: src/backend/services/users-api/src/app/routes.py [hunk 1]
```diff
@@ -24,122 +25,124 @@ def login():
...
+        "umnetID": umnetID.upper(),
+        "password": password,
+        "public_key": public_key
     }
 
     try:
         mongo.db.users.insert_one(user)
     except Exception as e:
-        return jsonify(err=FailureReturnString.DATABASE_VERIFICATION_FAILURE.value, error=str(e)), HttpCode.BAD_REQUEST.value
+        return jsonify(error=FailureReturnString.DATABASE_VERIFICATION_FAILURE.value, erroror=str(e)), HttpCode.BAD_REQUEST.value
```
Comment (madisonfines) [inline src/backend/services/users-api/src/app/routes.py]: Small spelling "error"!

File: src/backend/services/users-api/src/app/routes.py [hunk 2]
```diff
@@ -24,122 +25,124 @@ def login():
...
         if usr["password"] != curr_password:
             raise Exception("password verification failed")
-        
-        res = mongo.db.users.update_one({"umnetID" : umnetID.upper()}, {"$set" : user})
+
+        res = mongo.db.users.update_one(
+            {"umnetID": umnetID.upper()}, {"$set": user})
     except Exception as e:
-        return jsonify(err=FailureReturnString.DATABASE_VERIFICATION_FAILURE.value, error=str(e)), HttpCode.BAD_REQUEST.value
+        return jsonify(error=FailureReturnString.DATABASE_VERIFICATION_FAILURE.value, erroror=str(e)), HttpCode.BAD_REQUEST.value
```
Comment (madisonfines) [inline src/backend/services/users-api/src/app/routes.py]: Same here!

File: src/frontend/src/components/Button.vue [hunk 3]
```diff
@@ -1,29 +1,43 @@
...
 <script lang="ts">
 import { Component, Prop, Vue } from "vue-property-decorator";
+import Circle2 from "vue-loading-spinner/src/components/Circle2.vue";
 
-@Component
+@Component({
+  components: {
+    Circle2,
+  },
+  computed: {
```
Comment (madisonfines) [inline src/frontend/src/components/Button.vue]: Can probably move this to export default class then mark it as "get"!

File: src/frontend/src/components/Button.vue [hunk 4]
```diff
@@ -72,4 +86,8 @@ export default class Button extends Vue {
   transform: $hover-transform-button;
   box-shadow: $box-shadow-hover;
 }
+
+.loading {
+  margin-left: 25%;
```
Comment (madisonfines) [inline src/frontend/src/components/Button.vue]: Move this to style.scss please :)

File: src/frontend/src/pages/CreateAccountPage.vue [hunk 5]
```diff
@@ -0,0 +1,106 @@
+<template>
+  <div class="home">
+    <img
+      width="300px"
```
Comment (madisonfines) [inline src/frontend/src/pages/CreateAccountPage.vue]: You can move width and height to css by making this image have a class name. Also, I believe the login page has the same size logo so just use those variables from style for reuse!

File: src/frontend/src/pages/CreateAccountPage.vue [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+    };
+    this.$store.dispatch("ACTION_CREATE_ACCOUNT", values);
+  }
+}
+</script>
+
+<style lang="scss">
+@import "../style.scss";
+
+.home {
```
Comment (madisonfines) [inline src/frontend/src/pages/CreateAccountPage.vue]: Change this class name to create-account please!

File: src/frontend/src/pages/CreateAccountPage.vue [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+  text-align: center;
+  padding: 25px;
+  font-family: Avenir, Helvetica, Arial, sans-serif;
+  -webkit-font-smoothing: antialiased;
+  -moz-osx-font-smoothing: grayscale;
+}
+
+.login-button {
```
Comment (madisonfines) [inline src/frontend/src/pages/CreateAccountPage.vue]: Change this name to create account button so it doesn't clash with the login-button class in a different component. Also please move margins to style.scss!

File: src/frontend/src/pages/CreateAccountPage.vue [hunk 8]
```diff
@@ -0,0 +1,106 @@
...
+  margin-top: 100px;
+  margin-bottom: 50px;
+}
+
+body {
```
Comment (madisonfines) [inline src/frontend/src/pages/CreateAccountPage.vue]: You don't need this body section! Background gets set in App.vue

File: src/frontend/src/pages/CreateAccountPage.vue [hunk 9]
```diff
@@ -0,0 +1,106 @@
...
+  background-color: $background-color;
+}
+
+img {
```
Comment (madisonfines) [inline src/frontend/src/pages/CreateAccountPage.vue]: Please add relevant class names for img! When you use them like this it effects all images in the entire app so you wanna choose a specific class name

File: src/frontend/src/pages/LoginPage.vue [hunk 10]
```diff
@@ -5,25 +5,26 @@
...
       id="button"
-      dest="/home"
       label="Login"
+      dest=""
       size="small"
       type="default"
+      @click.native="login"
     />
-    <a class="create-account-link" href="TODO">Don't have an account? Create one now!</a>
+    <router-link to="/createAccount" id="create-account-link" tag="a"
```
Comment (madisonfines) [inline src/frontend/src/pages/LoginPage.vue]: Add class name to router-link so colour of link can be changed through class!

File: src/frontend/src/pages/CreateAccountPage.vue [hunk 11]
```diff
@@ -0,0 +1,106 @@
...
+  margin-bottom: 100px;
+  margin-top: 100px;
+}
+
+a {
```
Comment (madisonfines) [inline src/frontend/src/pages/CreateAccountPage.vue]: Don't need this link class!

File: src/frontend/src/pages/LoginPage.vue [hunk 12]
```diff
@@ -37,8 +38,23 @@ import Button from "../components/Button.vue";
     TextInput,
     Button,
   },
+  methods: {
```
Comment (madisonfines) [inline src/frontend/src/pages/LoginPage.vue]: Can move this method inside export default class

File: src/frontend/src/pages/NewTransactionPage.vue [hunk 13]
```diff
@@ -48,6 +48,7 @@ import Button from "../components/Button.vue";
 })
 export default class NewTransactionPage extends Vue {
   newTransaction() {
+
```
Comment (madisonfines) [inline src/frontend/src/pages/NewTransactionPage.vue]: Can remove this line!

File: src/frontend/tests/e2e/specs/testCreateAccount.js [hunk 14]
```diff
@@ -0,0 +1,38 @@
+describe("After loading the app", () => {
```
Comment (madisonfines) [inline src/frontend/tests/e2e/specs/testCreateAccount.js:1]: We should also be testing all the different toast messages the user can receive!

File: src/frontend/tests/e2e/specs/testWallet.js [hunk 15]
```diff
@@ -1,26 +1,28 @@
-describe('On the homepage', () => {
-  context('1080p resolution', () => {
+describe("On the homepage", () => {
```
Comment (madisonfines) [inline src/frontend/tests/e2e/specs/testWallet.js:1]: Same here! Check wallet related toast messages

**Review by madisonfines — CHANGES_REQUESTED** [review]

File: src/frontend/src/pages/CreateAccountPage.vue [hunk 16]
```diff
@@ -0,0 +1,104 @@
...
+
+.create-account {
+  text-align: center;
+  padding: $create-account-padding;
+  font-family: Avenir, Helvetica, Arial, sans-serif;
+  -webkit-font-smoothing: antialiased;
+  -moz-osx-font-smoothing: grayscale;
+}
+
+.login-logo {
```
Comment (madisonfines) [inline src/frontend/src/pages/CreateAccountPage.vue]: I don't even think you need this here, I believe if you make the logo's class login-logo it'll reference all the info from the LoginPage!

**Review by madisonfines — APPROVED** [review]
One small comment, but looks good to me!

dwight

**Review by abhisachdev17 — APPROVED** [review]
Looks good to me!

**Pull request merged**